.github/workflows/verible-ci.yml

```diff
@@ -69,20 +85,28 @@ jobs:
         format: 'YYYY-MM-DD-HH-mm-ss'
 
     - name: Retrieve cached results
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           /tmp/clang-tidy-hashes.cache
-          /home/runner/.cache/bazel
+          /root/.cache/bazel
         key: clang-tidy-${{ steps.cache_timestamp.outputs.time }}
         restore-keys: clang-tidy-
 
     - name: Run clang tidy
       run: ./.github/bin/run-clang-tidy.sh
 
+    - name: 📤 Upload performance graphs
+      uses: actions/upload-artifact@v2
+      if: success() || failure() 
```

Comment:
Couldn't we just use `always()`, or would that semantically be different ?

Reply:
It is different. As per last comment in the related [SO answer](https://stackoverflow.com/a/58859404), the `success() || failure() ` expression allows the user to cancel the build, while `always()` would insist on running the step upon cancellation.

Reply:
Ah, I see.